### DISCUSSION TITLE
Current transposed matrix return for quite a lot of objects.

### DISCUSSION THREAD

**Discussion by thomasmansencal** [colour-science/colour#688] [Ideas]
@MichaelMauderer I think it's important to address before the first release, the MPC legacy of the API and it's heavy use being involved with a lot of (3, 3) matrices computations result in the fact that a lot if not most of the colour models objects returns (3, 1) transposed matrices, for instance:

``` Python
    >>> colour.XYZ_to_xyY(np.array([0.1180583421, 0.1034, 0.0515089229]))
    array([[ 0.4325]
           [ 0.3788]
           [ 0.1034]])
```

I would be keen on getting rid of that and just output a normal (3, ) matrix as follows:

``` Python
    >>> colour.XYZ_to_xyY(np.array([0.1180583421, 0.1034, 0.0515089229]))
    array([0.4325,  0.3788, 0.1034])
```

What do you think?

**Comment by MichaelMauderer** [comment]
That actually did throw me off at one point.

General thoughts:
- I think it's okay if there is a good rational and it's well documented (e.g., does it make it easier to parallelize computations by passing arrays instead of numerics.
- I seem to remember something about varying conventions for indexing. Might be important to be consistent with the other scientific python libraries here.

Otherwise if there is no good reason to keep it, I'd agree with getting rid of it to simplify usage.

**Comment by thomasmansencal** [comment]
It made stuff easier when I was using the API with its private part for MPC but now, I don't see anything really useful about it. Let's get rid of that :)

Cheers,
